### DISCUSSION TITLE
Running the given example

### DISCUSSION THREAD

**Discussion by ProjectPool-ML** [explosion/spaCy#10292] [Help: Coding & Implementations]
Error with the example:
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-10-294d8953f0aa> in <module>
      5 visualizers = ["ner", "textcat"]
      6 print(models)
----> 7 spacy_streamlit.visualize(models, default_text, visualizers)
      8 

~\.conda\envs\venv-fromheathrow\lib\site-packages\spacy_streamlit\visualizer.py in visualize(models, default_text, default_model, visualizers, ner_labels, ner_attrs, similarity_texts, token_attrs, show_json_doc, show_meta, show_config, show_visualizer_select, show_pipeline_info, sidebar_title, sidebar_description, show_logo, color, key, get_default_text)
     94 
     95     if "parser" in visualizers and "parser" in active_visualizers:
---> 96         visualize_parser(doc, key=key)
     97     if "ner" in visualizers and "ner" in active_visualizers:
     98         ner_labels = ner_labels or nlp.get_pipe("ner").labels

~\.conda\envs\venv-fromheathrow\lib\site-packages\spacy_streamlit\visualizer.py in visualize_parser(doc, title, key)
    151         html = displacy.render(sent, options=options, style="dep")
    152         # Double newlines seem to mess with the rendering
--> 153         html = html.replace("\n\n", "\n")
    154         if split_sents and len(docs) > 1:
    155             st.markdown(f"> {sent.text}")

AttributeError: 'NoneType' object has no attribute 'replace'

**Comment by polm** [comment]
Sorry, I don't understand this.

Where did you get this example from? What are you trying to do? Please [format your code legibly](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) so we can read it.

**Reply by GmailBhu** [reply]
This is an example from the space website

**Reply by polm** [reply]
Uh, can you link to it? There are a lot of examples on the website and I don't recognize this one.

Also are you the same person?

**Reply by GmailBhu** [reply]
sorry about not being clear. The code I'm using is:

```
# streamlit_app.py
import spacy_streamlit

models = ["en_core_web_sm", "en_core_web_md"]
default_text = "Sundar Pichai is the CEO of Google."
spacy_streamlit.visualize(models, default_text)
```

The error I get is this:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-1-1444109b7920> in <module>
      4 models = ["en_core_web_sm", "en_core_web_md"]
      5 default_text = "Sundar Pichai is the CEO of Google."
----> 6 spacy_streamlit.visualize(models, default_text)

~\.conda\envs\venv-fromheathrow\lib\site-packages\spacy_streamlit\visualizer.py in visualize(models, default_text, default_model, visualizers, ner_labels, ner_attrs, similarity_texts, token_attrs, show_json_doc, show_meta, show_config, show_visualizer_select, show_pipeline_info, sidebar_title, sidebar_description, show_logo, color, key, get_default_text)
     94 
     95     if "parser" in visualizers and "parser" in active_visualizers:
---> 96         visualize_parser(doc, key=key)
     97     if "ner" in visualizers and "ner" in active_visualizers:
     98         ner_labels = ner_labels or nlp.get_pipe("ner").labels

~\.conda\envs\venv-fromheathrow\lib\site-packages\spacy_streamlit\visualizer.py in visualize_parser(doc, title, key)
    151         html = displacy.render(sent, options=options, style="dep")
    152         # Double newlines seem to mess with the rendering
--> 153         html = html.replace("\n\n", "\n")
    154         if split_sents and len(docs) > 1:
    155             st.markdown(f"> {sent.text}")

**AttributeError: 'NoneType' object has no attribute 'replace'**
```

Thank you

**Reply by polm** [reply]
It looks like you're using the example from the [spacy-streamlit README](https://github.com/explosion/spacy-streamlit). 

Did you install the English model first like it says in the README?

What version of everything do you have installed? You can check with `conda list --export`.
